src/libs/Localize/LocaleListener/BaseLocaleListener.js

```diff
@@ -0,0 +1,38 @@
+import Onyx from 'react-native-onyx';
+import CONST from '../../../CONST';
+import ONYXKEYS from '../../../ONYXKEYS';
+
+let preferredLocale = CONST.DEFAULT_LOCALE;
+
+/**
+ * Adds event listener for changes to the locale so Onyx gets updated accordingly
+ *
+ * @param {Function} callbackAfterChange
+ */
+const connect = (callbackAfterChange) => {
+    Onyx.connect({
+        key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+        callback: (val) => {
+            if (!val) {
+                return;
+            }
+
+            preferredLocale = val;
+            callbackAfterChange(val);
+        },
+    });
+};
+
+/*
+* @return {String}
+*/
+function getPreferredLocale() {
```

Comment:
Why wouldn't it work? It was working for me in the cases I was using it but I am not sure if it was just a coincidence because most of the behavior is handled by the listener either way.

Reply:
I take it back, this is fine :oh_nothing: